### PR TITLE
One box: Add search like code snippets

### PR DESCRIPTION
**Pull request by vovakulikov**
Part of [AI-205](https://linear.app/sourcegraph/issue/AI-205/show-code-snippets-for-context-files)

This PR 
- Adds a search file content code snippets block; it later will be used for search results within chat in one-box mode.
- Adds preview button to context item link which renders highlighted code snippet block (originally this PR was only about code snippet for one box, but I've found having code snippet preview for link very useful, links are usually very long and you don't remember which file it was, helped me to avoid opening a lot of tabs by checking just preview blocks)

Screenshot 2024-09-11 at 00 45 07

## Todo
- [x] Add an unhighlighted text block 
- [x] Add highlights support 
- [x] Add demo 
- [x] Align colors for different vscode themes 

## Test plan
- CI is green
- Check storybook demo 
- Run VSCode (or Cody Web) check that you can see preview code blocks next to file context files

### CODE REVIEW

**Comment by vovakulikov** [comment]
@thenamankumar I still need to go through colors design tokens but the logic is ready for the review

**Comment by vovakulikov** [comment]
@thenamankumar I finished the color token work and also supported highlights file fetching in this PR as well. This PR can and probably should go independently from https://github.com/sourcegraph/cody/pull/5199 

As soon as this is merged and code blocks will be on main we can easily add them for search mode results.

**Comment by vovakulikov** [comment]
Update from meeting with @thenamankumar 
- We will hide the Preview button behind the feature flag for now since we don't won't introduce new unpolished yet feature to the standard context items
- We will bring this back as soon as we get more high-fidelity designs  (after one box V0 PR is merged)

File: vscode/webviews/chat/cells/contextCell/ContextCell.tsx [hunk 1]
```diff
@@ -165,22 +164,7 @@ export const ContextCell: FunctionComponent<{
...
-                                                    isTooLarge={item.isTooLarge}
-                                                    isTooLargeReason={item.isTooLargeReason}
-                                                    isIgnored={item.isIgnored}
-                                                    className={clsx(
-                                                        styles.contextItem,
-                                                        MENTION_CLASS_NAME
-                                                    )}
-                                                    linkClassName={styles.contextItemLink}
-                                                />
+                                                <FileContextItem item={item} />
```
Comment (thenamankumar) [inline vscode/webviews/chat/cells/contextCell/ContextCell.tsx:167]: thanks for not spreading the item attributes for no reason.

**Review by thenamankumar — APPROVED** [review]

**Pull request merged**